src/oemof/solph/constraints/set_idle_time.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+
+"""Constraints to relate variables in an existing model.
+
+SPDX-FileCopyrightText: Jann Launer
+
+SPDX-License-Identifier: MIT
+"""
+from pyomo import environ as po
+
+
+def set_idle_time(model, f1, f2, n, name_constraint="constraint_idle_time"):
+    r"""
+    Adds a constraint to the given model that enforces f1 to be inactive
+    for n timesteps before f2 can be active.
+
+    For each timestep status of f2 can only be active if f1 has been inactive
+    the previous n timesteps.
+
+    **Constraints:**
+
+    .. math:: X_1(s) + X_2(t) <= 1 \forall 0 < t < n, 0 \le s \le t
+    .. math:: X_1(s) + X_2(t) <= 1 \forall t \ge n, t-n \le s \le t
+
+    Parameters
+    ----------
+    model : oemof.solph.Model
+        Model to which the constraint is added.
+    f1 : tuple
+        First flow tuple.
+    f2 : tuple
+        Second flow tuple. Has to be inactive for a defined number of
+        timesteps after first flow was active.
+    n : int
```

Comment:
This constraint would be more versatile if the idile time would be given as a time. If non-equidistant time steps are used, their number is not really mmeaningful.

Reply:
Good idea! However, I see it as an optional future development that can be addressed when it is needed.